src/app/zap-templates/zcl/data-model/chip/joint-fabric-datastore-cluster.xml

```diff
@@ -0,0 +1,298 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2024 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="DatastoreStateEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="Pending" value="0x00"/>
+    <item name="Committed" value="0x01"/>
+    <item name="DeletePending" value="0x02"/>
+    <item name="CommitFailed" value="0x03"/>
+  </enum>
+
+  <enum name="FailureCodeEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="OK" value="0x00"/>
+  </enum>
+
+  <struct name="DatastoreStatusEntry">
```

Comment:
This struct does not exist in the spec.

I am stopping here.  This XML seems to be loosely inspired by the spec, at best....  

So in order:

1. Make the spec make sense (right now it's doing things it's not allowed to do).
2. Make the XML actually follow the spec.

Reply:
This PR says it's based on https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10743, so that's what I was looking at.  There is no `DatastoreStatusEntry` type in the spec as of that PR.

> Could you kindly clarify what you mean by 'loosely inspired'?

It's sort of what the spec has, but has various random differences from it for no particular reason.